### PR TITLE
[MBL-1582] Run the app on Xcode 16 and iOS 18

### PR DESCRIPTION
**Pull request by jovaniks**
# 📲 What

- Update the App Center to `5.0.5`.
- Update the `UIApplicationType` protocol because the `open` function signature has been changed on Xcode 16.

# 🤔 Why

Updates were required to ensure the app runs smoothly on Xcode 16.

# 🛠 How

By updating the `App Center` dependency and modifying the `UIApplicationType` protocol, the app is now compatible with Xcode 16 and can run on iOS 18 devices.

# 👀 See

- [Jira](https://kickstarter.atlassian.net/browse/MBL-1582)
- [QA Script - iOS 18 upgrade](https://docs.google.com/spreadsheets/d/1ErhnhCqV2bfMJPh1M3EkZw0kfr5Pu5VHDQsTfzV11fU/edit?usp=sharing) 

| iPad  | iPhone 📱 |
| --- | --- |
| Screenshot 2024-09-17 at 10 17 29 AM | IMG_901FBDD04657-1 |

# ✅ Acceptance criteria

- [x] App can run on Xcode 16 and iOS 18
- [x] Issues found should be documented and tracked in Jira tickets

# ⏰ TODO

- Fix the issues found
- Circle CI updates

### CODE REVIEW

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/FormatTests.swift:528:1 | warning: (trailingSpace) Remove trailing space at end of a line. |
| Library/FormatTests.swift:533:1 | warning: (trailingSpace) Remove trailing space at end of a line. |
| Library/FormatTests.swift:538:1 | warning: (trailingSpace) Remove trailing space at end of a line. |
| Library/FormatTests.swift:543:1 | warning: (trailingSpace) Remove trailing space at end of a line. |
| Library/TestHelpers/MockApplication.swift:15:1 | warning: (trailingSpace) Remove trailing space at end of a line. |
| Library/TestHelpers/MockApplication.swift:19:1 | warning: (unusedArguments) Mark unused function arguments with _. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ TARGET ?= Kickstarter-Framework
 PLATFORM ?= iOS
 RELEASE ?= itunes
 # Keep simulator in sync with `Library/TestHelpers/TestCase.swift` and `.circleci/config.yml`
-IOS_VERSION ?= 17.2
+IOS_VERSION ?= 17.5
```
Comment (ankushkushwaha) [inline Makefile:9]: Is there any decision made by the core team to drop support for iOS 17.2 ?

Comment (ifosli) [reply Makefile:9]: What do you mean by drop support? We should definitely have the app still work for users on iOS 17.2. However, if we're updating our circleCI config, my recommendation is to have developers match that configuration locally.

File: .circleci/config.yml [hunk 2]
```diff
@@ -40,7 +41,9 @@ test_job: &test_job
         at: .
     - macos/preboot-simulator:
         # Keep simulator in sync with `Library/TestHelpers/TestCase.swift` and `Makefile`
-        version: "17.2"
+        # iOS 17.5 is minimum supported runtime by circle-ci -> 16.0 (16A242d)
+        # more info: https://circle-macos-docs.s3.amazonaws.com/image-manifest/v15048/manifest.txt
+        version: "17.5"
```
Comment (ifosli) [inline .circleci/config.yml:46]: I'm glad this didn't require re-recording snapshots - I was worried about that. Did you check if updating to 18.0 would also pass tests without any further work needed? If they're equally convenient, I think we may want to update to iOS 18 instead. (Optional; we could also update to iOS 18 later!)

File: Makefile (hunk 1 above)
Comment (jovaniks) [reply Makefile:9]: plus the @ifosli replied, the iOS 17.5 is the minimum runtime version available in CircleCI->Xcode 16 

more info https://circle-macos-docs.s3.amazonaws.com/image-manifest/v15048/manifest.txt

File: .circleci/config.yml (hunk 2 above)
Comment (jovaniks) [reply .circleci/config.yml:46]: Yup, I also tried with iOS 18 so definitely we need to re-recording the snapshots :/, do we need a migration plan?

**Review by stevestreza-ksr — APPROVED** [review]

**Review by scottkicks — APPROVED** [review]

**Comment by scottkicks** [comment]
@jovaniks  I'll let you know when the release is fully out!

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/TestHelpers/MockApplication.swift:17:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:18:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:19:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:20:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:21:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:22:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:23:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:25:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:26:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:27:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:28:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:29:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:30:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/TestHelpers/MockApplication.swift:31:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:7:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:8:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:9:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:10:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:11:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:13:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:14:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:15:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:16:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/UIApplicationType.swift:17:1 | warning: (indent) Indent code in accordance with the scope level. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by ankushkushwaha — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (ankushkushwaha) [reply .circleci/config.yml]: Thanks for clarification.

File: Makefile (hunk 1 above)
Comment (ankushkushwaha) [reply Makefile]: Thanks for clarification.

**Pull request merged**